reVX/handlers/xmission_cost_layers.py

```diff
@@ -13,3 +15,89 @@ class XmissionCostsLayers(ExclusionLayers):
     """
     Handler for Transmission cost layers
     """
+    def __init__(self, h5_file, capacity, hsds=False):
+        """
+        Parameters
+        ----------
+        h5_file : str | list | tuple
+            .h5 file containing exclusion layers and techmap,
+            or a list of h5 files
+        capacity : int
+            Capacity bin to extract costs for
+        hsds : bool, optional
+            Boolean flag to use h5pyd to handle .h5 'files' hosted on AWS
+            behind HSDS, by default None
+        """
+        self._capacity = capacity
+        super().__init__(h5_file, hsds=hsds)
+
+    def __getitem__(self, keys):
+        ds, ds_slice = parse_keys(keys)
+
+        if ds.lower().startswith('lat'):
```

Comment:
Understood you're trying to catch lat and latitude, but it also allows misspellings and other unexpected options to successfully return data.

Reply:
__getitem__ allows for flexibilty, its executed as self[...] and as you said we can allow for custom functions. here we are adding in the ability to "get" "costs" and "mcp_costs" by computing them on the fly